io-engine/src/grpc/v1/nexus.rs

```diff
@@ -1147,4 +1148,47 @@ impl NexusRpc for NexusService {
         })
         .await
     }
+
+    async fn list_rebuild_history(
+        &self,
+        request: Request<ListRebuildHistoryRequest>,
+    ) -> GrpcResult<RebuildHistoryMapResponse> {
+        let args = request.into_inner();
+        let count = args.count.unwrap_or_default();
+        let rx = rpc_submit::<_, _, nexus::Error>(async move {
+            let mut history_map: HashMap<String, RebuildHistoryResponse> =
+                HashMap::new();
+            for nexus in nexus::nexus_iter() {
+                let mut history_list: VecDeque<RebuildHistoryRecord> =
+                    VecDeque::new();
+                let nexus_uuid = nexus.uuid().to_string();
+                for record in
+                    nexus_lookup(&nexus_uuid)?.rebuild_history().iter()
+                {
+                    // We will return all of the rebuild records for nexus if
+                    // count is 0,Apply only non zero
+                    // limit.
+                    if history_list.len() == count.encoded_len() && count != 0 {
```

Comment:
If you need just to limit list size, why don't you use vec slice or an iterator helper method that could this more idiomatically?

Reply:
I wanted to remove first element specifically when list len reaches limit. This is because we push from the back so first element will be the older rebuild job. Hence thought of using VecDeque. With slice i can can form the whole list as is, but while truncating will i be able to give preference to objects at higher indexes?